x/distribution/client/cli/tx.go

```diff
@@ -34,7 +35,233 @@ const (
 	MaxMessagesPerTxDefault = 5
 )
 
+// NewTxCmd returns a root CLI command handler for all x/distribution transaction commands.
+func NewTxCmd(storeKey string, m codec.Marshaler, txg tx.Generator, ar tx.AccountRetriever) *cobra.Command {
+	distTxCmd := &cobra.Command{
+		Use:                        types.ModuleName,
+		Short:                      "Distribution transactions subcommands",
+		DisableFlagParsing:         true,
+		SuggestionsMinimumDistance: 2,
+		RunE:                       client.ValidateCmd,
+	}
+
+	distTxCmd.AddCommand(flags.PostCommands(
+		NewWithdrawRewardsCmd(m, txg, ar),
+		NewWithdrawAllRewardsCmd(m, txg, ar, storeKey),
+		NewSetWithdrawAddrCmd(m, txg, ar),
+		NewFundCommunityPoolCmd(m, txg, ar),
+	)...)
+
+	return distTxCmd
+}
+
+type newGenerateOrBroadcastFunc func(ctx context.CLIContext, txf tx.Factory, msgs ...sdk.Msg) error
+
+func newSplitAndApply(
+	newGenerateOrBroadcast newGenerateOrBroadcastFunc,
+	cliCtx context.CLIContext,
+	txBldr tx.Factory,
+	msgs []sdk.Msg,
+	chunkSize int,
+) error {
+	if chunkSize == 0 {
+		return newGenerateOrBroadcast(cliCtx, txBldr, msgs...)
+	}
+
+	// split messages into slices of length chunkSize
+	totalMessages := len(msgs)
+	for i := 0; i < len(msgs); i += chunkSize {
+
+		sliceEnd := i + chunkSize
+		if sliceEnd > totalMessages {
+			sliceEnd = totalMessages
+		}
+
+		msgChunk := msgs[i:sliceEnd]
+		if err := newGenerateOrBroadcast(cliCtx, txBldr, msgChunk...); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func NewWithdrawRewardsCmd(m codec.Marshaler, txg tx.Generator, ar tx.AccountRetriever) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "withdraw-rewards [validator-addr]",
+		Short: "Withdraw rewards from a given delegation address, and optionally withdraw validator commission if the delegation address given is a validator operator",
+		Long: strings.TrimSpace(
+			fmt.Sprintf(`Withdraw rewards from a given delegation address,
+and optionally withdraw validator commission if the delegation address given is a validator operator.
+
+Example:
+$ %s tx distribution withdraw-rewards cosmosvaloper1gghjut3ccd8ay0zduzj64hwre2fxs9ldmqhffj --from mykey
+$ %s tx distribution withdraw-rewards cosmosvaloper1gghjut3ccd8ay0zduzj64hwre2fxs9ldmqhffj --from mykey --commission
+`,
+				version.ClientName, version.ClientName,
+			),
+		),
+		Args: cobra.ExactArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			inBuf := bufio.NewReader(cmd.InOrStdin())
+			txf := tx.NewFactoryFromCLI(inBuf).
+				WithTxGenerator(txg).
+				WithAccountRetriever(ar)
+			cliCtx := context.NewCLIContextWithInputAndFrom(inBuf, args[0]).WithMarshaler(m)
+
+			delAddr := cliCtx.GetFromAddress()
+			valAddr, err := sdk.ValAddressFromBech32(args[0])
+			if err != nil {
+				return err
+			}
+
+			msgs := []sdk.Msg{types.NewMsgWithdrawDelegatorReward(delAddr, valAddr)}
+			if viper.GetBool(flagCommission) {
+				msgs = append(msgs, types.NewMsgWithdrawValidatorCommission(valAddr))
+			}
```

Comment:
validate msgs before broadcasting
```suggestion
			}
			
			for _, msg := range msgs {
				if err := msg.ValidateBasic(); err != nil {
					return err
				}
			}
``

Reply:
We don't do this for all command handlers, but we really should.

Reply:
yeah, this is a good idea. I'll add for all distr commands for now